src/Planner/PlannerJoinTree.cpp

```diff
@@ -963,7 +963,9 @@ JoinTreeQueryPlan buildQueryPlanForTableExpression(QueryTreeNodePtr table_expres
                             return true;
 
                         const auto join_kind = join_node->getKind();
-                        if (join_kind == JoinKind::Left || join_kind == JoinKind::Right || join_kind == JoinKind::Inner)
+                        const auto join_strictness = join_node->getStrictness();
+                        if (join_kind == JoinKind::Left || join_kind == JoinKind::Right
+                            || (join_kind == JoinKind::Inner && join_strictness == JoinStrictness::All))
```

Comment:
The check is similar to https://github.com/ClickHouse/ClickHouse/blob/95b21f4a44d0fe56924d2959ccf008add6ab59ff/src/Planner/findParallelReplicasQuery.cpp#L106

It wasn't consistent before